clang/include/clang/Basic/Attr.td

```diff
@@ -1768,6 +1768,30 @@ class SYCLAddIRAttrMemberCodeHolder<code Code> {
 
 // Common class for SYCL add_ir_attributes_* attributes.
 def SYCLAddIRAttrCommonMembers : SYCLAddIRAttrMemberCodeHolder<[{
+  static std::optional<std::string>
+  getValidAttributeNameAsString(const APValue &Value,
+                                 const ASTContext &Context,
+                                 QualType ValueQType) {
+    assert(!Value.isLValue());
+    if (ValueQType->isCharType()) {
+      char C = static_cast<char>(Value.getInt().getExtValue());
+      return std::string(&C, 1);
+    }
+    if (ValueQType->isArrayType() &&
+        (ValueQType->getArrayElementTypeNoTypeQual()->isCharType())) {
+      SmallString<10> StrBuffer;
+      for (unsigned I = 0; I < Value.getArrayInitializedElts(); ++I) {
+        const APValue &ArrayElem = Value.getArrayInitializedElt(I);
+        char C = static_cast<char>(ArrayElem.getInt().getExtValue());
```

Comment:
Should we not add an assert here to check that each array element is indeed an integer before casting it to a char? 
```suggestion
        const APValue &ArrayElem = Value.getArrayInitializedElt(I);
        assert(ArrayElem.isInt() && "Expected integer value for character type");
        char C = static_cast<char>(ArrayElem.getInt().getExtValue());
```

Reply:
We check that array element is `char` above, on line 1781, so I think that such assert is not necessary

Reply:
Thanks for the clarification!